kuka_kmr_iisy_eac_driver/src/robot_manager_node.cpp

```diff
@@ -0,0 +1,213 @@
+  this->registerStaticParameter<std::string>(
+    "controller_ip", "", kuka_drivers_core::ParameterSetAccessRights{false, false},
+    [this](const std::string &) { return true; });
+  this->registerStaticParameter<std::string>(
+    "robot_model", "kmr_iisy", kuka_drivers_core::ParameterSetAccessRights{false, false},
+    [this](const std::string & robot_model)
+    { return this->onRobotModelChangeRequest(robot_model); });
+}
+
+// TODO(Svastits): rollback in case of failures
+rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+MobileRobotManagerNode::on_activate(const rclcpp_lifecycle::State &)
+{
+  // Join observe thread, necessary if previous activation failed
+  terminate_ = false;
+  // Activate hardware interface
+  if (!kuka_drivers_core::changeHardwareState(
+        change_hardware_state_client_, robot_model_, State::PRIMARY_STATE_ACTIVE, 5000))
+  {
+    RCLCPP_ERROR(get_logger(), "Could not activate hardware interface");
+    return FAILURE;
+  }
+
+  // Workaround until controller_manager/jtc bug is fixed:
+  std::this_thread::sleep_for(std::chrono::milliseconds(20));
+
+  // Activate RT controller(s)
+  auto controllers = controller_handler_.GetControllersForMode(control_mode_);
+  controllers.push_back(kuka_drivers_core::CARTESIAN_POSE_BROADCASTER);
+  if (!kuka_drivers_core::changeControllerState(change_controller_state_client_, controllers, {}))
+  {
+    RCLCPP_ERROR(get_logger(), "Could not activate RT controllers");
+    this->on_deactivate(get_current_state());
+    return FAILURE;
+  }
+
+  RCLCPP_INFO(get_logger(), "Successfully activated controllers");
+
+  // Return failure if control is stopped while in state activating
+  if (terminate_)
+  {
+    RCLCPP_ERROR(get_logger(), "Error occurred during driver activation");
+    return FAILURE;
+  }
+
+  return SUCCESS;
+}
+
+rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+MobileRobotManagerNode::on_deactivate(const rclcpp_lifecycle::State &)
+{
+  // Deactivate hardware interface
+  // Deactivation was not stable with 2000 ms timeout
+  if (!kuka_drivers_core::changeHardwareState(
+        change_hardware_state_client_, robot_model_, State::PRIMARY_STATE_INACTIVE, 3000))
+  {
+    RCLCPP_ERROR(get_logger(), "Could not deactivate hardware interface");
+    return ERROR;
+  }
+
+  // Stop RT controllers
+  auto controllers = controller_handler_.GetControllersForMode(control_mode_);
+  controllers.push_back(kuka_drivers_core::CARTESIAN_POSE_BROADCASTER);
+  if (!kuka_drivers_core::changeControllerState(
+        change_controller_state_client_, {}, controllers, SwitchController::Request::BEST_EFFORT))
+  {
+    RCLCPP_ERROR(get_logger(), "Could not stop RT controllers");
+    return ERROR;
+  }
+
+  RCLCPP_INFO(get_logger(), "Successfully stopped controllers");
+  return SUCCESS;
+}
+
+bool MobileRobotManagerNode::onControlModeChangeRequest(int control_mode)
```

Comment:
this driver only supports 1 control mode, so ideally this parameter would be removed.
But I think you can leave this for now, theoretically there is a possibiliy that a new control mode is added and you should not spend effort on this removal, so this is just a remark